toolchain/check/check.cpp

```diff
@@ -148,21 +155,72 @@ static auto TrackImport(
       return;
     }
 
-    // Check for explicit imports of the same library. The ID comparison is okay
-    // in this case because both come from the same file.
-    auto packaging = unit_info.unit->parse_tree->packaging_directive();
-    if (packaging && import.package_id == packaging->names.package_id &&
-        import.library_id == packaging->names.library_id) {
+    // True if the file's package is implicitly `Main` (by omission).
+    bool is_file_implicit_main =
+        !packaging || !packaging->names.package_id.is_valid();
+    // True if the import is using "current package" syntax (by omission).
+    bool is_import_current_package = !import.package_id.is_valid();
```

Comment:
What do you think about calling this `implicit_package` to make it more clear that it's testing for the implicit syntax, not just package equality?

Something like:
```suggestion
    // True if the import is using the implicit "current package" syntax
    // (by omitting an explicit package).
    bool is_import_implicit_package = !import.package_id.is_valid();
```

(not blocking either way)

Reply:
I've adjusted the comments, with a corresponding edit for is_package_implicit_main.

Renamed to `is_import_implicit_current_package`

The problem I have with `is_import_implicit_package` is that it's ambiguous in this context: `is_package_implicit_main` is right above, but `is_import_implicit_package` doesn't necessarily mean the implicit package of `Main`. I'd thought `current_package` was a balance of brevity and clarity, but since it doesn't work for  you, I've gone for the more verbose version.